dsp/window/window_test.go

```diff
@@ -8,170 +8,317 @@ import (
 	"fmt"
 	"testing"
 
+	"gonum.org/v1/gonum/cmplxs"
 	"gonum.org/v1/gonum/floats"
-	"gonum.org/v1/gonum/floats/scalar"
 )
 
+const tol = 1e-6
 const (
 	gaussWin = iota
 	tukeyWin
 )
 
-var windowTests = []struct {
-	name    string
-	fn      func([]float64) []float64
-	fnCmplx func([]complex128) []complex128
-	want    []float64
+// these types describe zero, mono, and multi parameter windowing
+// functions for both real and complex data
+//
+// when looking at the truth tables, it is helpful to know that
+// Go treats []complex128{1} as [1+0i].  More specifically, a variable which
+// is of type complex128 to which you assign an untyped constant has a zero
+// imaginary part.
```

Comment:
I don't think this is necessary. The interface definitions say the first part and the second part is described in the language spec.

Reply:
The second part is not part of [the spec](https://golang.org/ref/spec#Complex_numbers), at least not explicitly.  I also think that the most useful comments are the ones that explain something subtle about code, and this is a subtle behavior that you all but certainly will not _unambiguously_ be able to predict without seeing it before.

Reply:
The [spec explicitly shows that a `1` can be converted to a `comple128`](https://golang.org/ref/spec#Conversions); that's what is happening here.